datafusion/expr/src/registry.rs

```diff
@@ -27,9 +27,15 @@ use std::sync::Arc;
 
 /// A registry knows how to build logical expressions out of user-defined function' names
 pub trait FunctionRegistry {
-    /// Set of all available udfs.
+    /// Returns names of all available scalar user defined functions.
     fn udfs(&self) -> HashSet<String>;
 
+    /// Returns names of all available aggregate user defined functions.
+    fn udafs(&self) -> HashSet<String>;
+
+    /// Returns names of all available window user defined functions.
+    fn udwfs(&self) -> HashSet<String>;
```

Comment:
I just wonder should make this methods with default implementation (empty set) for 50.1 release and revert it like this for 51.0